### PR TITLE
TS-88-refactor-Rooms-Page

### LINKED ISSUE #88: Refactor: RoomPage and divide it into smaller components
Refactor RoomPage and all components which belong to it.

Acceptance Criteria:
The RoomPage is divided into smaller components and  its services are rewritten to sagas.

Sub Tasks \ Bugs:
 
- [x] divide RoomPage into smaller components ;

- [x]  rewrite services to sagas.

Estimate: 4 days

### PR DESCRIPTION
**Pull request by ViktoriiaKosovan**

### CODE REVIEW

File: frontend/src/components/RoomsPage/RoomsPage.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        if (isDisabled) {
+            getListOfDisabledRooms();
+        } else {
+            getListOfRooms();
+        }
+    }, [isDisabled]);
+
+    const submitRoomForm = (values) => {
+        const type = roomTypes.find((roomType) => roomType.id === values.type);
+        handleRoomFormSubmit({ ...values, type });
```
Comment (Niknikson) [inline frontend/src/components/RoomsPage/RoomsPage.js:53]: this is better to do in saga

File: frontend/src/components/RoomsPage/RoomsPage.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        setOpenConfirmDialog(true);
+    };
+
+    const changeGroupDisabledStatus = (currentId) => {
+        const foundRoom = [...disabledRooms, ...rooms].find(
+            (roomItem) => roomItem.id === currentId,
+        );
+        return isDisabled
+            ? toggleRoomVisibility({ ...foundRoom, disable: false }, isDisabled)
+            : toggleRoomVisibility({ ...foundRoom, disable: true }, isDisabled);
```
Comment (Niknikson) [inline frontend/src/components/RoomsPage/RoomsPage.js]: this shouldn't be here

File: frontend/src/components/RoomsPage/RoomsPage.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        if (deleteLabel === cardType.TYPE) {
+            deleteRoomType(selectedId);
+        }
+        if (deleteLabel === cardType.ROOM) {
+            if (confirmDialogType !== dialogTypes.DELETE_CONFIRM) {
+                changeGroupDisabledStatus(selectedId);
+            } else {
+                deleteRoom(selectedId, isDisabled);
+            }
+        }
```
Comment (Niknikson) [inline frontend/src/components/RoomsPage/RoomsPage.js]: switch case?

File: frontend/src/components/RoomsPage/RoomsPage.js (hunk 1 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/RoomsPage/RoomsPage.js:53]: done

File: frontend/src/components/RoomsPage/RoomsPage.js (hunk 2 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/RoomsPage/RoomsPage.js]: done

File: frontend/src/components/RoomsPage/RoomsPage.js (hunk 3 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/RoomsPage/RoomsPage.js]: done

**Review by HiroFive — APPROVED** [review]

**Review by Yashchemskyi-Oleksandr — APPROVED** [review]

**Pull request closed without merging**